packages/apps/reputation-oracle/server/src/modules/cron-job/cron-job.service.spec.ts

```diff
@@ -1,830 +1,369 @@
-      cronJobEntity.cronJobType = cronJobType;
-      cronJobEntity.completedAt = new Date();
-
-      const findOneByTypeSpy = jest
-        .spyOn(repository, 'findOneByType')
-        .mockResolvedValue(cronJobEntity);
+    it('should return false if the last cron job is completed', async () => {
+      const completedCronJob = new CronJobEntity();
+      completedCronJob.completedAt = new Date();
+      cronJobRepository.findOneByType.mockResolvedValue(completedCronJob);
 
-      const result = await service.isCronJobRunning(cronJobType);
+      const result = await service.isCronJobRunning(
+        CronJobType.ProcessPendingIncomingWebhook,
+      );
 
-      expect(findOneByTypeSpy).toHaveBeenCalledWith(cronJobType);
-      expect(result).toEqual(false);
+      expect(cronJobRepository.findOneByType).toHaveBeenCalledWith(
+        CronJobType.ProcessPendingIncomingWebhook,
+      );
+      expect(result).toBe(false);
     });
 
-    it('should return true if last cron job is not completed', async () => {
-      const cronJobType = CronJobType.ProcessPendingIncomingWebhook;
-      const cronJobEntity = new CronJobEntity();
-      cronJobEntity.cronJobType = cronJobType;
+    it('should return true if the last cron job is not completed', async () => {
+      const runningCronJob = new CronJobEntity();
+      cronJobRepository.findOneByType.mockResolvedValue(runningCronJob);
 
-      const findOneByTypeSpy = jest
-        .spyOn(repository, 'findOneByType')
-        .mockResolvedValue(cronJobEntity);
+      const result = await service.isCronJobRunning(
+        CronJobType.ProcessPendingIncomingWebhook,
+      );
 
-      const result = await service.isCronJobRunning(cronJobType);
-      expect(findOneByTypeSpy).toHaveBeenCalledWith(cronJobType);
-      expect(result).toEqual(true);
+      expect(cronJobRepository.findOneByType).toHaveBeenCalledWith(
+        CronJobType.ProcessPendingIncomingWebhook,
+      );
+      expect(result).toBe(true);
     });
   });
 
   describe('completeCronJob', () => {
     it('should complete a cron job', async () => {
-      const cronJobType = CronJobType.ProcessPendingIncomingWebhook;
       const cronJobEntity = new CronJobEntity();
-      cronJobEntity.cronJobType = cronJobType;
-
-      const mockDate = new Date(2023, 12, 23);
-
-      jest.useFakeTimers();
-      jest.setSystemTime(mockDate);
-
-      const updateOneSpy = jest
-        .spyOn(repository, 'updateOne')
-        .mockResolvedValue(cronJobEntity);
+      const completedCronJob = { ...cronJobEntity, completedAt: new Date() };
+      cronJobRepository.updateOne.mockResolvedValue(completedCronJob as any);
 
       const result = await service.completeCronJob(cronJobEntity);
 
-      expect(updateOneSpy).toHaveBeenCalled();
-      expect(cronJobEntity.completedAt).toEqual(mockDate);
-      expect(result).toEqual(cronJobEntity);
-
-      jest.useRealTimers();
+      expect(cronJobRepository.updateOne).toHaveBeenCalledWith(
```

Comment:
This `expect` is flaky now (I got it failed to 1ms when running on local), please consider to keep fake timers or use `expect.objectContaining`